File: backend/Entities/Ticket.cs
namespace Cinema.Entities;

public enum TicketStatus
{
    Available,
    Booked,
    Paid,
    Canceled,
    Unavailable
}

public class Ticket
{
    public int TicketId { get; set; }

    public string SeatId { get; set; }

    public string ScheduleId { get; set; }

    public double Cost { get; set; }

    public string? UserId { get; set; }

    public TicketStatus Status { get; set; } = TicketStatus.Available;
}

File: frontend/src/types/Ticket.ts
export const enum TicketStatus
{
    Available,  // Vừa được tạo
    Sold,       // Đâ bán chưa thanh toán
    Paid,       // Đã bán và thanh toán
    Canceled,   // Đã hủy
    Unavailable // Bị lỗi nên hủy
}

export default interface Ticket {
    id: number;
    ticketId: number,
    seatId: string,         // Tham chiếu ghế của vé này
    scheduleId: string,     // Tham chiếu lịch chiếu/suất chiếu
    cost: number,           // Giá vé
    userId: string|null,    // Người đã mua vé
    status: TicketStatus    // Trạng thái hiện tại của vé
}
